pkg/models/tagged_html.go

```diff
@@ -0,0 +1,207 @@
+package models
+
+import (
+	"context"
+	"database/sql/driver"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"regexp"
+
+	"github.com/samber/lo"
+
+	"github.com/cms-enterprise/easi-app/pkg/appcontext"
+	"github.com/cms-enterprise/easi-app/pkg/sanitization"
+)
+
+var (
+	spanRe      = regexp.MustCompile(`<span[^>]*>.*?</span>`)
+	attributeRe = regexp.MustCompile(`(\S+)="([^"]+)"`)
+)
+
+// TaggedHTML Is the input type for HTML that could contain tags
+type TaggedHTML TaggedContent
+
+// TaggedContent represents rich text HTML with possible tagged HTML
+type TaggedContent struct {
+	RawContent HTML
+	Tags       []*Tag
+}
+
+// UnmarshalGQLContext unmarshals the data from graphql to the TaggedHTML type
+func (th *TaggedHTML) UnmarshalGQLContext(_ context.Context, v interface{}) error {
+	rawHTML, ok := v.(string)
+	if !ok {
+		return errors.New("invalid TaggedHTML")
+	}
+
+	tc, err := NewTaggedContentFromString(rawHTML)
+	if err != nil {
+		return err
+	}
+	*th = TaggedHTML(tc)
+
+	return nil
+}
+
+// MarshalGQLContext marshals the TaggedHTML type to JSON to return to graphQL
+func (th TaggedHTML) MarshalGQLContext(ctx context.Context, w io.Writer) error {
+	logger := appcontext.ZLogger(ctx)
+
+	jsonValue, err := json.Marshal(th.RawContent)
+	if err != nil {
+		logger.Info("invalid TaggedHTML")
+		return fmt.Errorf("failed to marshal TaggedHTMLto JSON: %w", err)
+	}
+
+	if _, err := w.Write(jsonValue); err != nil {
+		return fmt.Errorf("failed to write TaggedHTML to writer: %w", err)
+	}
+
+	return nil
+}
+
+func (th TaggedHTML) UniqueTags() []*Tag {
+	uniqueTags := lo.UniqBy(th.Tags, func(tag *Tag) string {
+
+		key := fmt.Sprint(tag.TagType, tag.EntityRaw) //The entity raw, and tag type will be unique.
+		return key
+	})
+
+	return uniqueTags
+}
+
+// ToTaggedContent casts the input to TaggedContent
+func (th TaggedHTML) ToTaggedContent() TaggedContent {
+	return TaggedContent(th)
+}
+
+// NewTaggedContentFromString converts a rawString into TaggedHTMl. It will store the input string as the raw content,
+// and then sanitize and parse the input.
+func NewTaggedContentFromString(htmlString string) (TaggedContent, error) {
+	tags, err := tagsFromStringRegex(htmlString)
```

Comment:
is there a reason the string is not getting sanitized?

Reply:
updated!